src/API/src/db/bionomics/entities/environment.entity.ts

```diff
@@ -0,0 +1,75 @@
+import { Entity, Column, OneToOne } from 'typeorm';
+import { ObjectType, Field } from '@nestjs/graphql';
+import { BaseEntity } from '../../base.entity';
+import { Bionomics } from './bionomics.entity';
+
+@Entity('environment')
+@ObjectType({ description: 'bionomics environment data' })
+export class Environment extends BaseEntity {
+  @Column('varchar', { length: 50, nullable: true })
+  @Field({ nullable: true })
+  roof: string;
+
+  @Column('varchar', { length: 50, nullable: true })
+  @Field({ nullable: true })
+  walls: string;
+
+  @Column('boolean', { nullable: true })
+  @Field({ nullable: true })
+  house_screening: boolean;
+
+  @Column('boolean', { nullable: true })
+  @Field({ nullable: true })
+  open_eaves: boolean;
+
+  @Column('varchar', { length: 50, nullable: true })
+  @Field({ nullable: true })
+  cooking: string;
+
+  @Column('varchar', { length: 50, nullable: true })
```

Comment:
For notes fields we need to leave `length` blank, as that then sets it as the max length.